### DISCUSSION TITLE
How to call a relative address?

### DISCUSSION THREAD

**Discussion by marcussacana** [icedland/iced#341] [Q&A] [answered]
I'm trying catch the RIP in the code with this:
```asm
call +0 ;0xE8 00 00 00 00
pop RAX
```
Therefore if I use the `Instruction.CreateBranch` the assembler will take my '0' address as a absolute memory address;
then is there a way to assembly a call instruction with a relative address?

Notice that I can't use labels in the code without rewrite my tool, for me, should be more easy just manually put the 0xE8 00 00 00 00 in the memory after encode instead rewrite.

**Comment by wtfsck** [comment, marked as answer]
If you're using the Assembler C# class you can just call `call(Label)` and put the label after the call instruction. But since you say you can't use labels, you can just add the bytes yourself with `db(0xE8,0,0,0,0)`

**Reply by marcussacana** [reply]
db is a good idea, thx
